core/trino-spi/src/main/java/io/trino/spi/block/FixedBinaryBlockEncoding.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.block;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+
+import static io.trino.spi.block.EncoderUtil.decodeNullBits;
+import static io.trino.spi.block.EncoderUtil.encodeNullsAsBits;
+
+public class FixedBinaryBlockEncoding
+        implements BlockEncoding
+{
+    public static final String NAME = "FIXED_BINARY";
+
+    @Override
+    public String getName()
+    {
+        return NAME;
+    }
+
+    @Override
+    public void writeBlock(BlockEncodingSerde blockEncodingSerde, SliceOutput sliceOutput, Block block)
+    {
+        int positionCount = block.getPositionCount();
+        sliceOutput.appendInt(positionCount);
+        FixedBinaryBlockOutput blockOutput = (FixedBinaryBlockOutput) block;
+        int positionWidth = blockOutput.getPositionWidth();
+        sliceOutput.appendInt(positionWidth);
+
+        encodeNullsAsBits(sliceOutput, block);
+
+        if (!block.mayHaveNull()) {
+            sliceOutput.writeBytes(blockOutput.getValuesSlice());
+        }
+        else {
+            long[] valuesWithoutNull = new long[positionCount * positionWidth];
+            int nonNullPositionCount = 0;
+            for (int i = 0; i < positionCount; i++) {
+                for (int j = 0; j < positionWidth; j++) {
+                    valuesWithoutNull[nonNullPositionCount + j] = block.getLong(i, Long.BYTES * j);
```

Comment:
you could add another block method, e.g:
```
block.copyPositionTo(long[] longArray, int offset);
```
this way we could avoid repeated calls to `getLong`.